src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ObjectWriter.cs

```diff
@@ -326,7 +326,15 @@ public void BuildDebugLocInfoMap(ObjectNode node)
                 {
                     foreach (var loc in locs)
                     {
-                        _offsetToDebugLoc.Add(loc.NativeOffset, loc);
+                        if (_offsetToDebugLoc.ContainsKey(loc.NativeOffset) == false)
```

Comment:
`_offsetToDebugLoc.Add[loc.NativeOffset] = loc;` would be faster (avoids the redundant hash lookup). Unlike Add, the indexer does not throw if the key exists already.